video/decode/vd_lavc.h

```diff
@@ -0,0 +1,28 @@
+#ifndef MPLAYER_VD_LAVC
+#define MPLAYER_VD_LAVC
+
+#include <stdbool.h>
+
+struct vd_lavc_params {
```

Comment:
This is hidden as private for a reason. You can add a function that acts if any of the setting has changed. And check in command.c `opt_ptr == &opts->vd_lavc_params` only.

This is hidden to avoid leaking implementation details over other modules.

Reply:
So instead of the struct the header defines a function that takes the pointer and context and then does the `if` from command.c in there?

Reply:
function also doesn't fit into the design looking at it. 

If fact common way is to use flag like `flags & UPDATE_VD` and reinit then. Mark option with this flag.

Reply:
That's much better, thanks.

It looks to me that we could rename `UPDATE_HWDEC` to `UPDATE_VD` and do the same thing for any of those options instead of only the hwdec one.
Doing that required changing the absolute seek to a relative one, otherwise it wouldn't work well with vd-queue (starting a video would offset the starting position by vd-queue-secs seconds).